runtimes/dotnet-sdk-csharp.nix

```diff
@@ -0,0 +1,69 @@
+{pkgs, piston, ...}:
+let
+    pkg = pkgs.dotnet-sdk;
+in piston.mkRuntime {
+    language = "csharp";
+    version = pkg.version;
+    runtime = "dotnet-sdk";
+
+    aliases = [
+        "csharp.net"
+        "c#"
+        "cs"
+        "c#.net"
+        "cs.net"
+        "c#-dotnet"
+        "cs-dotnet"
+        "csharp-dotnet"
+        "dotnet-c#"
+        "dotnet-cs"
+        "dotnet-csharp"
+
+    ];
+
+    extraBuildCommands = ''
+        # Cache nuget packages in the pkg directory
+        cd ${pkg}
+        export DOTNET_CLI_HOME=$PWD
+        ./dotnet new console -o cache_application
+        rm -rf cache_application
+    '';
+
```

Comment:
Why cant we do this in the nix-store? That way its in a r/o state and we can recreate it much easier.
I think there is a better way to set this app up than doing a nuget restore.

Also, what your doing wont work because `${pkg}` is going to be in a read-only directory.

Reply:
How I suggest this be done is that you create a new derivation and run these commands within it.
That way you can use the restore command below with `${nugetPackages}/.nuget/packages` for example.
Nix will manage all the dependencies stuff for you - no need to specially tag it anywhere, just include it in the compile script.

Reply:
I did the same imementation we had last time. We created the .nuget directory in the read-only package directory while building. It works since extraBuildCommands is executed with cp.execSync not with a new user group. And while compiling and running, we only read from it. 

I thought about the derivation, but I think it required source code to be downloaded and built using packages in buildInput. I will look more into it.

Reply:
> We created the .nuget directory in the read-only package directory while building.
I'm pretty sure nix wont allow what your extraBuildCommands system does as the entirety of `/nix/store` is mounted as read-only by design, only allowing nix derivations to write into the store.

Reply:
The user in execSync has all permissions. I have tested it before pushing.

Reply:
It is bad practice to modify the nix-store.

Reply:
I managed to put the NuGet packages in a separate derivation. Turns out there are flags that determine the derivation phases. It is a better approach now.